test/functional/apps/discover/_shared_links.js

```diff
@@ -20,6 +20,11 @@ bdd.describe('shared links', function describeIndexTests() {
 
   bdd.before(function () {
     baseUrl = PageObjects.common.getHostPort();
```

Comment:
Are you suggesting that instead of stripping port `:80` or `:443` from baseUrl, that it could test to see if that was the port and if so, `baseUrl = common.getHost()` ?